src/skin/qml/qmleffectslotproxy.cpp

```diff
@@ -57,30 +55,19 @@ QString QmlEffectSlotProxy::getGroup() const {
 }
 
 QString QmlEffectSlotProxy::getEffectId() const {
-    const EffectPointer pEffect = m_pEffectSlot->getEffect();
-    if (!pEffect) {
-        return QString();
-    }
-
-    const EffectManifestPointer pManifest = pEffect->getManifest();
-    return pManifest->id();
+    return m_pEffectSlot->id();
 }
 
 void QmlEffectSlotProxy::setEffectId(const QString& effectId) {
-    m_pRack->maybeLoadEffect(
-            m_pChainSlot->getChainSlotNumber(),
-            m_pEffectSlot->getEffectSlotNumber(),
-            effectId);
+    const EffectManifestPointer pManifest =
+            m_pEffectsManager->getBackendManager()->getManifestFromUniqueId(
+                    effectId);
```

Comment:
This could be cleaned up by refactoring this class to store the EffectManifest instead of passing QStrings around to identify effects.

Reply:
Let's save that for a future PR.